### PR TITLE
Alerting Docs: Update the introduction to Templates

### PR DESCRIPTION
**Pull request by ppcano**
Updates [Introduction > Notifications > Templates](https://grafana.com/docs/grafana/latest/alerting/fundamentals/notifications/templates/) to simplify understanding of key templating concepts. 

Main changes include:
- Updated the diagram
- Refined the introduction key points
- Adjusted the three templating sections

More related changes to other templating docs will be included in upcoming PRs.

### CODE REVIEW

**Review by brendamuir — APPROVED** [review]

File: docs/sources/alerting/fundamentals/notifications/templates.md [hunk 1]
```diff
@@ -19,146 +19,167 @@ labels:
...
+1. The Alertmanager uses the contact point's notification template to format the message, then sends the notification to the configured destination(s)—an email address.
 
-        The host {{ index $labels "instance" }} has exceeded 80% CPU usage for the last 5 minutes
+## Template annotations
 
-  The outcome of this template would print:
+[Annotations](ref:annotations) can be defined in the alert rule to add extra information to alert instances.
 
-        The host instance 1 has exceeded 80% CPU usage for the last 5 minutes
+When creating an alert rule, Grafana suggests several optional annotations, such as `description`, `summary`, `runbook_url`, `dashboardUId` and `panelId`, which help identify and respond to alerts. You can also create custom annotations.
```
Comment (tonypowa) [inline docs/sources/alerting/fundamentals/notifications/templates.md:91]:
```suggestion
When creating an alert rule, Grafana suggests several optional annotations, such as `description`, `summary`, `runbook_url`,  or a `custom annotation`, which help identify and respond to alerts. You can also create custom annotations.
```

Comment (tonypowa) [reply docs/sources/alerting/fundamentals/notifications/templates.md:91]: `dashboardUId` and `panelId` not currently present in recent Grafana

File: docs/sources/alerting/fundamentals/notifications/templates.md [hunk 2]
```diff
@@ -19,146 +19,167 @@ labels:
...
 
-Notification templates represent the alternative approach to templating designed for reusing templates. Notifications are messages to inform users about events or conditions triggered by alerts. You can create reusable notification templates to customize the content and format of alert notifications. Variables, labels, or other context-specific details can be added to the templates to dynamically insert information like metric values.
+- Notification templates are assigned to the **Contact point**, rather than the alert rule.
+- If not specified, the contact point uses a default template that includes relevant alert information.
+- You can create reusable notification templates and reference them in other templates.
+- The same template can be shared across multiple contact points, making it easier to maintain and ensuring consistency.
+- While both annotation/label templates and notification templates use the same templating language, the available variables and functions differ. For more details, refer to the [notification template reference](ref:notification-message-reference) and [annotation/label template reference](ref:templating-labels-annotations).
```
Comment (tonypowa) [inline docs/sources/alerting/fundamentals/notifications/templates.md:151]: Really good point . And we should give it even more visibility. We can do that in a future PR

File: docs/sources/alerting/fundamentals/notifications/templates.md [hunk 3]
```diff
@@ -19,146 +19,167 @@ labels:
...
+   - Template notifications when you want to customize the appearance and information of your notifications.
+   - Avoid using notification templates to add extra information to alert instances—use annotations instead.
 
-{{<admonition type="note">}}
-Think about templating labels when you need to improve or change how alerts are uniquely identified. This is especially helpful if the labels you get from your query aren't detailed enough. Keep in mind that it's better to keep long sentences for summaries and descriptions. Also, avoid using the query's value in labels because it may result in the creation of many alerts when you actually only need one.
-{{</admonition>}}
+This diagram illustrates the entire templating process, from querying labels and templating the alert summary and notification to the final alert notification message.
 
-Templating can be applied by using variables and functions. These variables can represent dynamic values retrieved from your data queries.
+{{< figure src="/media/docs/alerting/how-notification-templates-works.png" max-width="1200px" caption="How templating works" >}}
```
Comment (tonypowa) [inline docs/sources/alerting/fundamentals/notifications/templates.md:77]: ❤️

**Review by tonypowa — APPROVED** [review]
LGTM!

**Pull request merged**